pmoves/services/messaging-gateway/main.py

```
@@ -0,0 +1,305 @@
+
+async def _nats_resilience_loop() -> None:
+    """Maintain resilient NATS connection with auto-reconnect."""
+    global _nc
+    backoff = 1.0
+
+    while True:
+        nc = NATS()
+        disconnect_event = asyncio.Event()
+
+        async def _disconnected_cb():
+            if not disconnect_event.is_set():
```

Ruff (0.14.8)

Function definition does not bind loop variable `disconnect_event`

(B023)